src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getWorkspaceState } from '../../../../core/server/utils';
+import {
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsCheckConflictsObject,
+  OpenSearchDashboardsRequest,
+  SavedObjectsFindOptions,
+} from '../../../../core/server';
+
+type WorkspaceOptions = Pick<SavedObjectsBaseOptions, 'workspaces'> | undefined;
+
+export class WorkspaceIdConsumerWrapper {
+  private formatWorkspaceIdParams<T extends WorkspaceOptions>(
+    request: OpenSearchDashboardsRequest,
+    options?: T
+  ): T {
+    const { workspaces, ...others } = options || {};
+    const workspaceState = getWorkspaceState(request);
+    const workspaceIdParsedFromRequest = workspaceState?.id;
+    const workspaceIdsInUserOptions = options?.workspaces;
+    let finalWorkspaces: string[] = [];
+    if (options?.hasOwnProperty('workspaces')) {
+      finalWorkspaces = workspaceIdsInUserOptions || [];
+    } else if (workspaceIdParsedFromRequest) {
+      finalWorkspaces = [workspaceIdParsedFromRequest];
+    }
+
+    return {
+      ...(others as T),
+      ...(finalWorkspaces.length ? { workspaces: finalWorkspaces } : {}),
+    };
+  }
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    return {
+      ...wrapperOptions.client,
+      create: <T>(type: string, attributes: T, options: SavedObjectsCreateOptions = {}) =>
+        wrapperOptions.client.create(
+          type,
+          attributes,
+          this.formatWorkspaceIdParams(wrapperOptions.request, options)
+        ),
+      bulkCreate: <T = unknown>(
+        objects: Array<SavedObjectsBulkCreateObject<T>>,
+        options: SavedObjectsCreateOptions = {}
+      ) =>
+        wrapperOptions.client.bulkCreate(
+          objects,
+          this.formatWorkspaceIdParams(wrapperOptions.request, options)
+        ),
+      checkConflicts: (
+        objects: SavedObjectsCheckConflictsObject[] = [],
+        options: SavedObjectsBaseOptions = {}
+      ) =>
+        wrapperOptions.client.checkConflicts(
+          objects,
+          this.formatWorkspaceIdParams(wrapperOptions.request, options)
+        ),
+      delete: wrapperOptions.client.delete,
+      find: (options: SavedObjectsFindOptions) =>
+        wrapperOptions.client.find(this.formatWorkspaceIdParams(wrapperOptions.request, options)),
+      bulkGet: wrapperOptions.client.bulkGet,
+      get: wrapperOptions.client.get,
+      update: wrapperOptions.client.update,
+      bulkUpdate: wrapperOptions.client.bulkUpdate,
+      addToNamespaces: wrapperOptions.client.addToNamespaces,
+      deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
```

Comment:
Shall we add `deleteByWorkspace` here?

Reply:
Sure, done for that. But as we workspace is a required param for deleteByWorkspace so I won't wrap the method.